### PR TITLE
Adding more SIMD constant folding support

### PR DESCRIPTION
**Pull request by tannergooding**
This adds support for other primitive unary and binary operations but does not cover relational operators yet.

Tests still need to be added as well.

### CODE REVIEW

**Comment by tannergooding** [comment]
spmi-replay failures are https://github.com/dotnet/runtime/issues/82397

**Comment by tannergooding** [comment]
CC. @dotnet/jit-contrib 

More SIMD constant folding support, covering the other basic binary operators (but not comparisons). All of these have equivalent scalar handling.

File: src/coreclr/jit/simd.h [hunk 1]
```diff
@@ -268,6 +301,119 @@ void EvaluateUnarySimd(genTreeOps oper, bool scalar, var_types baseType, TSimd*
...
+            unreached();
+        }
+    }
+}
+
+template <>
+inline float EvaluateBinaryScalarSpecialized<float>(genTreeOps oper, float arg0, float arg1)
+{
+    uint32_t arg0Bits = *reinterpret_cast<uint32_t*>(&arg0);
+    uint32_t arg1Bits = *reinterpret_cast<uint32_t*>(&arg1);
```
Comment (jakobbotsch) [inline src/coreclr/jit/simd.h:401]: This is technically UB -- can we use `memcpy` to be safe?

Comment (tannergooding) [reply src/coreclr/jit/simd.h:401]: Will update. Notably we're doing this in many places throughout the JIT, so its a more general thing that should be fixed (separately, if it's a concern).

**Review by jakobbotsch — APPROVED** [review]
LGTM!

**Pull request merged**